### PR TITLE
Add create service principal script to make it easy to generate.

### PR DESCRIPTION
**Pull request by wangtt03**
This change is [Reviewable](https://reviewable.io/reviews/azure/acs-engine/399)

### CODE REVIEW

**Comment by wangtt03** [comment]
Add a script to generate service principal so that users can easily generate it.
* script `create_service_principal.sh` is the basic functional script to generate.
* script `create_service_principal_docker.sh` is the docker wrapper to package the dependencies.

@colemickens Could you please review this PR? Thanks!

File: scripts/create_service_principal.sh [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+my_app_name_uuid=$(python -c 'import uuid; print (str(uuid.uuid4())[:8])')
+MY_APP_NAME="app${my_app_name_uuid}"
+
+MY_APP_KEY=$(python -c 'import uuid; print (uuid.uuid4().hex)')
+
+my_app_id_URI="${MY_APP_NAME}_id"
+
+#check if the user has subscriptions. If not she's probably not logged in
+subscriptions_list=$(azure account list --json)
```
Comment (rjtsdl) [inline scripts/create_service_principal.sh]: Can you change it to use az cli ? 
https://github.com/Azure/azure-cli

For this sentense, it is like `az account list --json`

File: scripts/create_service_principal.sh [hunk 2]
```diff
@@ -0,0 +1,139 @@
+#!/bin/bash
+
+#echo "Usage:
+#  1 bash create-service-principal.sh
+#  2 bash create-service-principal.sh <Subscription ID>
+# You need Azure CLI: https://docs.microsoft.com/azure/xplat-cli-install
```
Comment (rjtsdl) [inline scripts/create_service_principal.sh]: I recommend to use az cli, instead of xplat-cli-install. 
Check, https://github.com/Azure/azure-cli

**Comment by colemickens** [comment]
This is a lot of code to merge and maintain when a simple `az login` and `az ad sp create-for-rbac` is sufficient and outputs nearly the same information.

Further, it uses the deprecated azure-xplat-cli, rather than the recently GA'd new azure-cli.

**Comment by wangtt03** [comment]
@colemickens it is simple to create sp using CLI 2.0 so I will close this PR. Thanks!

**Pull request closed without merging**